ArduPlane/systemid.cpp

```diff
@@ -0,0 +1,294 @@
+            break;
+        case AxisType::INPUT_ROLL:
+            attitude_offset_deg.x = waveform_sample;
+            break;
+        case AxisType::INPUT_PITCH:
+            attitude_offset_deg.y = waveform_sample;
+            break;
+        case AxisType::INPUT_YAW:
+            attitude_offset_deg.z = waveform_sample;
+            break;
+        case AxisType::RECOVER_ROLL:
+            attitude_offset_deg.x = waveform_sample;
+            attitude_control->bf_feedforward(false);
+            break;
+        case AxisType::RECOVER_PITCH:
+            attitude_offset_deg.y = waveform_sample;
+            attitude_control->bf_feedforward(false);
+            break;
+        case AxisType::RECOVER_YAW:
+            attitude_offset_deg.z = waveform_sample;
+            attitude_control->bf_feedforward(false);
+            break;
+        case AxisType::RATE_ROLL:
+            attitude_control->rate_bf_roll_sysid(radians(waveform_sample));
+            break;
+        case AxisType::RATE_PITCH:
+            attitude_control->rate_bf_pitch_sysid(radians(waveform_sample));
+            break;
+        case AxisType::RATE_YAW:
+            attitude_control->rate_bf_yaw_sysid(radians(waveform_sample));
+            break;
+        case AxisType::MIX_ROLL:
+            attitude_control->actuator_roll_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_PITCH:
+            attitude_control->actuator_pitch_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_YAW:
+            attitude_control->actuator_yaw_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_THROTTLE:
+            throttle_offset += waveform_sample;
+            break;
+    }
+
+    // reduce control in XY axis when in position controlled modes
+    plane.quadplane.pos_control->set_xy_control_scale_factor(xy_control_mul);
+
+    log_data();
+}
+
+// @LoggerMessage: SIDD
+// @Description: System ID data
+// @Field: TimeUS: Time since system startup
+// @Field: Time: Time reference for waveform
+// @Field: Targ: Current waveform sample
+// @Field: F: Instantaneous waveform frequency
+// @Field: Gx: Delta angle, X-Axis
+// @Field: Gy: Delta angle, Y-Axis
+// @Field: Gz: Delta angle, Z-Axis
+// @Field: Ax: Delta velocity, X-Axis
+// @Field: Ay: Delta velocity, Y-Axis
+// @Field: Az: Delta velocity, Z-Axis
+
+// log system id
+void AP_SystemID::log_data() const
+{
+#if HAL_LOGGING_ENABLED
+    Vector3f delta_angle;
+    float delta_angle_dt;
+    plane.ins.get_delta_angle(delta_angle, delta_angle_dt);
+
+    Vector3f delta_velocity;
+    float delta_velocity_dt;
+    plane.ins.get_delta_velocity(delta_velocity, delta_velocity_dt);
+
+    if (is_positive(delta_angle_dt) && is_positive(delta_velocity_dt)) {
+        const float dt_ang_inv = 1.0 / delta_angle_dt;
+        const float dt_vel_inv = 1.0 / delta_velocity_dt;
+        AP::logger().WriteStreaming("SIDD", "TimeUS,Time,Targ,F,Gx,Gy,Gz,Ax,Ay,Az",
```

Comment:
There might be a argument for this to not be streaming, if you drop a few logs it will mess up the the analysis.

Reply:
that won't help, as the other messages needed are streaming too